### PR TITLE
Use validators file to stake, unstake, etc

### PR DESCRIPTION
**Pull request by popenta**
Added a parameter '--validators-file' for staking operations like **remove-nodes, stake-nodes, unbond-nodes, unstake-nodes** and **unjail-nodes**.

The commands can be used by providing the bls keys using `--bls-keys` or the validators file using `--validators-file`.

### CODE REVIEW

File: multiversx_sdk_cli/cli_delegation.py [hunk 1]
```diff
@@ -209,7 +216,16 @@ def stake_nodes(args: Any):
     cli_shared.send_or_simulate(tx, args)
 
 
+def _check_if_either_bls_keys_or_validators_file_are_provided(args: Any):
+    bls_keys = args.bls_keys
+    validators_file = args.validators_file
+
+    if not bls_keys and not validators_file:
+        raise errors.BadUsage("No bls keys or validators file provided")
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/cli_delegation.py:224]: Optionally, can be changed to _do this, do that_: "Either specify the ..., or the ...".

File: pyproject.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ build-backend = "hatchling.build"
 
 [project]
 name = "multiversx-sdk-cli"
-version = "9.4.1"
+version = "9.5.0"
```
Comment (andreibancioiu) [inline pyproject.toml:7]: :+1:

File: multiversx_sdk_cli/tests/test_cli_staking_provider.py [hunk 3]
```diff
@@ -72,7 +73,45 @@ def test_remove_nodes(capsys: Any):
     assert transaction["gasLimit"] == 13645500
 
 
-def test_stake_nodes(capsys: Any):
+def test_remove_nodes_with_validators_file(capsys: Any):
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/tests/test_cli_staking_provider.py:76]: :rocket:

File: multiversx_sdk_cli/delegation/staking_provider.py [hunk 4]
```diff
@@ -102,7 +108,13 @@ def prepare_transaction_for_removing_nodes(self, owner: IAccount, args: Any) ->
...
     def prepare_transaction_for_staking_nodes(self, owner: IAccount, args: Any) -> ITransaction:
         delegation_contract = Address.new_from_bech32(args.delegation_contract)
-        public_keys = self._parse_public_bls_keys(args.bls_keys)
+
+        if args.bls_keys:
+            public_keys = self._parse_public_bls_keys(args.bls_keys)
+        else:
+            validators_file_path = Path(args.validators_file).expanduser()
+            validators_file = ValidatorsFile(validators_file_path)
+            public_keys = validators_file.load_public_keys()
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/delegation/staking_provider.py:117]: Can it be extracted to `_load_validator_public_keys(args)` (duplicated below)?

File: multiversx_sdk_cli/validators/validators_file.py [hunk 5]
```diff
@@ -34,6 +35,19 @@ def load_signers(self) -> List[ValidatorSigner]:
...
 
+    def load_public_keys(self) -> List[ValidatorPublicKey]:
+        public_keys: List[ValidatorPublicKey] = []
+
+        for validator in self.get_validators_list():
+            # Get path of "pemFile", make it absolute
+            validator_pem = Path(validator.get("pemFile")).expanduser()
+            validator_pem = validator_pem if validator_pem.is_absolute() else self.validators_file_path.parent / validator_pem
+
+            pem_file = ValidatorPEM.from_file(validator_pem)
```
Comment (andreibancioiu) [inline multiversx_sdk_cli/validators/validators_file.py:46]: These 3 lines can also be extracted to a separate function e.g. `load_validator_pem(path): ValidatorPEM` / optional (code exists above, as well).

File: multiversx_sdk_cli/cli_delegation.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk_cli/cli_delegation.py:224]: Changed

File: multiversx_sdk_cli/delegation/staking_provider.py (hunk 4 above)
Comment (popenta) [reply multiversx_sdk_cli/delegation/staking_provider.py:117]: Extracted the code in a method

File: multiversx_sdk_cli/validators/validators_file.py [hunk 6]
```diff
@@ -34,6 +35,19 @@ def load_signers(self) -> List[ValidatorSigner]:
 
         return signers
 
+    def load_public_keys(self) -> List[ValidatorPublicKey]:
+        public_keys: List[ValidatorPublicKey] = []
+
+        for validator in self.get_validators_list():
+            # Get path of "pemFile", make it absolute
+            validator_pem = Path(validator.get("pemFile")).expanduser()
```
Comment (ssd04) [inline multiversx_sdk_cli/validators/validators_file.py:43]: it should be clear for the user, the format of the validators file, that if should have `pemFile`. Maybe to document it somehow in README or in command help, or to make reference to the example file that we have

File: multiversx_sdk_cli/validators/validators_file.py (hunk 5 above)
Comment (popenta) [reply multiversx_sdk_cli/validators/validators_file.py:46]: extracted in a method

File: multiversx_sdk_cli/validators/validators_file.py (hunk 6 above)
Comment (popenta) [reply multiversx_sdk_cli/validators/validators_file.py:43]: I agree. As of now, it is documented in the [docs](https://docs.multiversx.com/validators/staking/#the-staking-transaction), but in the future maybe we'll add a dedicated section in the cookbook.

**Review by andreibancioiu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**